src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2846,11 +2846,17 @@ ReplicaReservations::no_reply_t::no_reply_t(
   auto timeout =
     conf.get_val<std::chrono::milliseconds>("osd_scrub_reservation_timeout");
 
-  m_abort_callback = new LambdaContext([this, now_is]([[maybe_unused]] int r) {
+  m_abort_callback = new LambdaContext(
+             [this, now_is, pgid = parent.m_pg->pg_id]([[maybe_unused]] int r) {
+    PGRef pg = m_osds->osd->lookup_lock_pg(pgid);
```

Comment:
You shouldn't need to use lookup_lock_pg here, capture a reference directly.

Reply:
I think you also need to check the epoch against the current interval after locking the PG.